README.md

```diff
@@ -1 +1,1148 @@
-# balena-cli-masterclass
\ No newline at end of file
+balenaCLI Masterclass
+=====================
+
+**Masterclass Type:** Core
+
+# Introduction
+
+The balena Command Line Interface (balenaCLI) utility consists of a number
+of commands that allow a user to develop, deploy and manage balena applications
+and devices.
+
+Everything that can be achieved via the balenaCloud dashboard can also be
+achieved via the balenaCLI.
+
+In this masterclass, you will learn how to:
+
+* Login to your account
+* Push application code to a balena Application
+* Deploy locally built code to a balena Application
+* SSH into a balena device
+* Push and build applications on a device on the local network for fast
+    development and prototyping
+* Use private Docker registries and modules from a balena device
+
+# Hardware and Software Requirements
+
+It is assumed that the reader has access to the following:
+
+* A [balenaFin 1.1](https://store.balena.io/collections/developer-kit/products/balenafin-v1-1-0-developer-kit)
+* A suitable text editor for developing code on your development platform (eg.
+    [Visual Code](https://code.visualstudio.com/))
+* A suitable shell environment for command execution (such as `bash`)
+* A [balenaCloud](https://www.balena.io/) account
+* A familiarity with [Dockerfiles](https://docs.docker.com/engine/reference/builder/)
+* A [NodeJS](https://nodejs.org/en/) installation, including [NPM](https://www.npmjs.com/get-npm)
+
+# Exercises
+
+All of the following exercises assume that you are running the balenaCLI from
+a suitable Unix based shell. The exercises include commands which can be run
+in such a shell, and are represented by a line prefixed with `$`. Information
+returned from execution of a command may be appended under the line to show
+what might be returned. For example:
+
+```
+$ balena version
+11.9.3
+```
+
+## 1. Installation and Authentication
+
+First we need to install balenaCLI. Open a terminal on your development machine
+and run the following command:
+```
+$ npm --global install balena-cli
```

Comment:
A general question about maintainability: should we default/redirect/generate these instructions from the CLI instructions, rather than maintaining a fork here? Paulo and I noticed recently some out of date installation instructions in the OS marketing docs, so it's at the top of my mind.

On the other hand, I like the idea of this masterclass being self contained.

Reply:
When I started writing the `README.md` I started putting links to other documents in. Then I started going through the class as if I was a first time reader. Once I'd got five tabs open, I suddenly realised this was not going to end well, and that only concentrating on one document was far more preferable. At this point, I switched out the links to instructions in the document (and of course, this was the first change ;-) ).

What I *really* want is what we've been discussing for ages, which is pulling in specific snippets of other partial documents to generate a new document. In that case, we'd specify the partial for installing `balena-cli` in the README partial which would get regenerated. Unfortunately, we're nowhere near that yet. So for the moment, it's forked.

Reply:
Actually, what I *really* want is something that can generate both this document *and* generate a fully interactive online tutorial. It would be so useful for all the masterclasses, especially if we could hook it into our backend and our devices. But that's not happening aaaanytime soon.